src/plugins/ReplaceGoogleSearch/index.tsx

```diff
@@ -0,0 +1,126 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { definePluginSettings } from "@api/Settings";
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+import { Flex, Menu } from "@webpack/common";
+
+interface Engine {
+    name: string;
+    url: string;
+}
+
+const DefaultEngines = {
+    Google: "https://www.google.com/search?q=",
+    DuckDuckGo: "https://duckduckgo.com/",
+    Bing: "https://www.bing.com/search?q=",
+    Yahoo: "https://search.yahoo.com/search?p=",
+    Github: "https://github.com/search?q=",
+    Kagi: "https://kagi.com/search?q=",
+    Yandex: "https://yandex.com/search/?text=",
+    AOL: "https://search.aol.com/aol/search?q=",
+    Baidu: "https://www.baidu.com/s?wd=",
+    Wikipedia: "https://wikipedia.org/w/index.php?search=",
+} as const;
+
+const settings = definePluginSettings({
+    customEngineName: {
+        description: "Name of the custom search engine",
+        type: OptionType.STRING,
+        placeholder: "Google"
+    },
+    customEngineURL: {
+        description: "The URL of your Engine",
+        type: OptionType.STRING,
+        placeholder: "https://google.com/search?q="
+    }
+});
+
+function search(src: string, engine: string) {
+    open(engine + encodeURIComponent(src), "_blank");
+}
+
+export function makeSearchItem() {
+    const src = document.getSelection()?.toString();
+    if (!src) return;
+
+    const Engines = { ...DefaultEngines };
+
+    if(settings.store.customEngineName && settings.store.customEngineURL) {
+        Engines[settings.store.customEngineName] = settings.store.customEngineURL;
+    }
+
+    return (
+        <Menu.MenuItem
+            label="Search Text with..."
+            key="search-text"
+            id="search-text"
+        >
+            {Object.keys(Engines).map((engine, i) => {
+                const key = "search-content-" + engine;
+                if (!key) return;
+                return (
+                    <Menu.MenuItem
+                        key={key}
+                        id={key}
+                        label={
+                            <Flex style={{ alignItems: "center", gap: "0.5em" }}>
+                                <img
+                                    style={{
+                                        borderRadius: i >= 3
+                                            ? "50%"
+                                            : void 0
+                                    }}
+                                    aria-hidden="true"
+                                    height={16}
+                                    width={16}
+                                    src={`https://www.google.com/s2/favicons?domain=${new URL(Engines[engine])}`}
```

Comment:
what does this URL constructor achieve?
```suggestion
                                    src={`https://www.google.com/s2/favicons?domain=${Engines[engine]}`}
```